rueidislimiter/limit.go

```diff
@@ -0,0 +1,21 @@
+package rueidislimiter
+
+import "time"
+
+type RateLimit struct {
+	limit  int
+	window time.Duration
+}
+type RateLimitOption func(*RateLimit)
+
+func WithLimit(limit int) RateLimitOption {
+	return func(rl *RateLimit) {
+		rl.limit = limit
+	}
+}
+
+func WithWindow(window time.Duration) RateLimitOption {
+	return func(rl *RateLimit) {
+		rl.window = window
+	}
+}
```

Comment:
```suggestion
type RateLimit struct {
	limit  int
	window time.Duration
}
type RateLimitOption func(RateLimit) RateLimit

func WithLimit(limit int) RateLimitOption {
	return func(rl RateLimit) RateLimit {
		rl.limit = limit
		return rl
	}
}

func WithWindow(window time.Duration) RateLimitOption {
	return func(rl RateLimit) RateLimit {
		rl.window = window
		return rl
	}
}
```

This can have better performance and fewer allocations.

Reply:
This one looks good to me. Thanks @altanozlu!